pybossa/view/projects.py

```diff
@@ -1252,8 +1255,28 @@ def task_presenter(short_name, task_id):
         "user_id": user_id
     }
 
+    if task_submitter_id is not None:
+        taskruns = task_repo.filter_task_runs_by(task_id=task_id,
+                                                 project_id=project.id)
+        taskruns_dict = {tr.user_id: tr.info for tr in taskruns}
```

Comment:
It's possible that tasks has many contributors. Is it possible to optimize this call so that it fetches task runs by user id, project_id, task_id from db and return. Or build a new interface to support this.

Reply:
Great suggestion. I tested locally and I can pass user_id to filter_task_runs_by. Will update it.